app/views/member/_table.html.erb

```diff
@@ -0,0 +1,51 @@
+<% if members.length() == 0 %>
+<p>There are no members yet!</p>
+<% else %>
+
+
+<table class="mt-3 table listing" summary="Service list">
+<tr class="header">
+  <th>Name</th>
+  <th>Email</th>
+  <th>PP</th>
+  <th>Missed Mandatory</th>      
+  <th>Hours</th>
+</tr>
+<% members.each do |member| %>
+<tr>
+  <td><%= member.name %></td>
+  <td><%= member.email %></td>
+
+  <td>
+  <% if @past_events.length() == 0 %>
+        No events have occured!
+        <% else %>
+        <%= ((member.events.length().to_f / @past_events.length()) * 100.to_f).round(2) %>%
```

Comment:
preferable to move business logic out of HTML

Reply:
Completed this by providing functions within member.rb